src/api/draftSpecs.ts

```diff
@@ -215,21 +189,36 @@ const CHUNK_SIZE = 10;
 export const deleteDraftSpecsByCatalogName = async (
     draftId: string,
     specType: Entity,
-    catalogNames: string[]
+    catalogNames: string[],
+    operation: 'remove' | 'preserve'
 ) => {
     // In case we get an absolutely massive amount of catalogs to delete,
     // we don't want to spam supabase
     const limiter = pLimit(3);
     const promises: Array<Promise<PostgrestResponse<any>>> = [];
     let index = 0;
 
-    const deletePromiseGenerator = (idx: number) =>
-        supabaseClient
+    const deletePromiseGenerator = (idx: number) => {
+        let queryBuilder = supabaseClient
             .from(TABLES.DRAFT_SPECS)
             .delete()
             .eq('draft_id', draftId)
-            .eq('spec_type', specType)
-            .in('catalog_name', catalogNames.slice(idx, idx + CHUNK_SIZE));
+            .eq('spec_type', specType);
+
+        queryBuilder =
```

Comment:
When the `operation` parameter has the value `'remove'` the array of catalog names indicate an entities that should be deleted from the draft.

When the `operation` parameter has the value `'preserve'` the array of catalog names indicate an entities that should remain in the draft, deleting all unmatched, `draft_specs` entries of the indicated type from the draft.